api-operator/pkg/controller/api/api_controller.go

```diff
@@ -365,23 +386,29 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
 		// create Kaniko job
 		// if updating api or overriding api or image not found
 		if instance.Spec.UpdateTimeStamp != "" || instance.Spec.Override || !imageExist {
+			r.recorder.Event(instance, corev1.EventTypeNormal, "Configs",
+				"Handling analytics & interceptors, rendering dockerfile & mgw configs, and creating the Kaniko job.")
 			// handling analytics
 			log.Info("Handling analytics")
 			if err := analytics.Handle(&r.client, userNamespace); err != nil {
+				r.recorder.Event(instance, eventTypeError, "Configs", "Error while handling analytics.")
 				return reconcile.Result{}, err
 			}
 
 			// handling interceptors
 			log.Info("Handling interceptors")
 			if err := interceptors.Handle(&r.client, instance); err != nil {
 				log.Error(err, "Error handling interceptors")
+				r.recorder.Event(instance, eventTypeError, "Configs", "Error while handling interceptors.")
 				return reconcile.Result{}, err
 			}
 
 			// handling Kaniko docker file
 			log.Info("Rendering the dockerfile for Kaniko job and adding volumes to the Kaniko job")
 			if err := kaniko.HandleDockerFile(&r.client, userNamespace, instance.Name, ownerRef); err != nil {
 				log.Error(err, "Error rendering the docker file for Kaniko job and adding volumes to the Kaniko job")
+				r.recorder.Event(instance, eventTypeError, "KanikoJob",
+					"Error rendering the dockerfile for kaniko job.")
```

Comment:
Shouldn't the k be capitalized in 'Error rendering the dockerfile for **kaniko** job'

Reply:
Those have been removed in the latest code.